app/src/main/java/com/keylesspalace/tusky/settings/AccountPreferenceDataStore.kt

```diff
@@ -1,15 +1,21 @@
 package com.keylesspalace.tusky.settings
 
 import androidx.preference.PreferenceDataStore
+import com.keylesspalace.tusky.appstore.EventHub
 import com.keylesspalace.tusky.appstore.PreferenceChangedEvent
 import com.keylesspalace.tusky.db.AccountEntity
 import com.keylesspalace.tusky.db.AccountManager
+import com.keylesspalace.tusky.di.ApplicationScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class AccountPreferenceHandler(
-    private val account: AccountEntity,
+class AccountPreferenceDataStore @Inject constructor(
     private val accountManager: AccountManager,
-    private val dispatchEvent: (PreferenceChangedEvent) -> Unit
+    private val eventHub: EventHub,
+    @ApplicationScope val externalScope: CoroutineScope
```

Comment:
why is it called `external`?

Reply:
That's from https://developer.android.com/topic/architecture/data-layer#make_an_operation_live_longer_than_the_screen, where the examples there (and in the accompanying blog post at https://medium.com/androiddevelopers/coroutines-patterns-for-work-that-shouldnt-be-cancelled-e26c40f142ad) use the name `externalScope`.

The naming choice is not properly explained, but I think the intent is to communicate that although this is a (coroutine) scope, the code that uses it should not make assumptions about the lifetime of the scope.

So it can't be assumed to be a particular `lifecycleScope`, or `viewModelScope`, `GlobalScope`, etc. Just one that the code that created this object knows will last long enough for the operations.

If it's injected then it will be an `@ApplicationScope`, but if the object is constructed "by hand" then it could be any scope.